### PR TITLE
check other cluster role bindings for subscription admin

### PR DESCRIPTION
**Pull request by rokej**
Signed-off-by: Roke Jung <roke@redhat.com>

Check other cluster role bindings to determine subscription admin role.

### CODE REVIEW

File: pkg/utils/gitrepo.go [hunk 1]
```diff
@@ -999,6 +1006,44 @@ func matchUserSubAdmin(client client.Client, userIdentity, userGroups string) bo
...
+	if err == nil {
+		for _, binding := range bindingList.Items {
+			if binding.RoleRef.Kind == "ClusterRole" && binding.RoleRef.Name == "open-cluster-management:subscription-admin" {
+				klog.Infof("Found cluster role binding %s with open-cluster-management:subscription-admin cluster role.", binding.Name)
+
+				for _, subject := range binding.Subjects {
+					if strings.Trim(subject.Name, "") == strings.Trim(userIdentity, "") && strings.Trim(subject.Kind, "") == "User" {
+						klog.Info("User match. cluster-admin: true")
+
+						isUserSubAdmin = true
```
Comment (philipwu08) [inline pkg/utils/gitrepo.go:1025]: Possible enhancement is to return true when we know the user is a sub admin and stop iterating through the loop

**Comment by rokej** [comment]
The test failure is not related to this change. Merging.

**Pull request merged**